ticdc: add note to the Avro protocol (#19573)
pingcap/docs: ticdc/ticdc-avro-protocol.md

> - For Delete events, Avro only encodes the Key part. The Value part is empty. 
> - For Insert events, Avro encodes all column data to the Value part. 
> - For Update events, Avro encodes only all column data that is updated to the Value part.

> >   

Reviewer: ```suggestion
> 
```

> > If you want the old values with Update or Delete events, you can use other protocols.

Reviewer 2: ```suggestion
> The Avro protocol does not encode the old values for Update and Delete events. Additionally, in order to be compatible to most Confluent sink connectors which only recognizes `null` record for deletion (`"delete.on.null"`), a Delete event cannot carry extension information such as `_tidb_commit_ts` even when enabling `enable-tidb-extension`. If you need these features, consider using other protocols such as Canal-JSON or Debezium.
```

> > The Avro protocol does not encode the old values for Update and Delete events. Additionally, to be compatible with most Confluent sink connectors that only recognize a `null` record for deletions (`"delete.on.null"`), Delete events do not include extension information, such as `_tidb_commit_ts`, even when `enable-tidb-extension` is enabled. If you need these features, consider using other protocols such as Canal-JSON or Debezium.

Reviewer: ```suggestion
> The Avro protocol does not encode the old values for Update and Delete events. Additionally, to be compatible with most Confluent sink connectors that rely on `null` records to identify deletions (`delete.on.null`), Delete events do not include extension information, such as `_tidb_commit_ts`, even when `enable-tidb-extension` is enabled. If you need these features, consider using other protocols such as Canal-JSON or Debezium.
```

> > The Avro protocol does not encode the old values for Update and Delete events. Additionally, in order to be compatible to most Confluent sink connectors which only recognizes `null` record for deletion (`"delete.on.null"`), a Delete event cannot carry extension information such as `_tidb_commit_ts` even when enabling `enable-tidb-extension`. If you need these features, consider using other protocols such as Canal-JSON or Debezium.

Reviewer: ```suggestion
> The Avro protocol does not encode the old values for Update and Delete events. Additionally, to be compatible with most Confluent sink connectors that only recognize a `null` record for deletions (`"delete.on.null"`), Delete events do not include extension information, such as `_tidb_commit_ts`, even when `enable-tidb-extension` is enabled. If you need these features, consider using other protocols such as Canal-JSON or Debezium.
```
